haystack/dataclasses/document.py

```diff
@@ -114,7 +119,8 @@ def _create_id(self):
         mime_type = self.blob.mime_type if self.blob is not None else None
         meta = self.meta or {}
         embedding = self.embedding if self.embedding is not None else None
-        data = f"{text}{dataframe}{blob}{mime_type}{meta}{embedding}"
+        sparse_embedding = self.sparse_embedding.to_dict() if self.sparse_embedding is not None else ""
```

Comment:
This differs a bit from the other ones to not alter the id of existing Documents.
I can change it if you think it's better.

Reply:
Good catch! This approach looks good to me 👍